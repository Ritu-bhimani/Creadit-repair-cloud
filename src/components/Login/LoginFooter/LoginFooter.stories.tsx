import { ComponentMeta, ComponentStory } from '@storybook/react';
import { LoginFooter } from './LoginFooter';

export default {
  title: 'Login Footer',
  component: LoginFooter
} as ComponentMeta<typeof LoginFooter>;

const Template: ComponentStory<typeof LoginFooter> = (args) => (
  <LoginFooter {...args} />
);

export const Primary = Template.bind({});

Primary.args = {
  trainingLink: ''
};
