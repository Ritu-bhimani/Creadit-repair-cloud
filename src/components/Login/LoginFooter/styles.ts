import { css } from '@emotion/react';

export const LoginFooterStyles = {
  root: css`
    text-align: center;
  `,
  button: css`
    box-sizing: border-box;
    margin-top: 10px;
    display: inline-block;
    max-width: 338px;
    width: 90%;
    background: #fc115d;
    background: #fc115d;
    color: #fff;
    text-decoration: none;
    padding: 11px 0;
    font-size: 18px;
    :hover{
      text-decoration:underline;
      background-color:#e52d70;
    }
  `,
  bottomText: css`
    > span {
      border-right: 1px solid #5b5b5b;
      margin-right: 2px;
      padding-right: 5px;
    }
    > img {
      margin-right: 6px;
    }
    bottom: 20px;
    left: 0;
    right: 0;
    margin: 1vh auto;
    color: #5b5b5b;
    line-height: 20px;
    background: 0 0;
    text-align: center;
  `,
  justGettingStartedText: css`
    margin: 20px 0 0;
    font-size: 16px;
    line-height: 22px;
    color: #5b5b5b;
    letter-spacing: 0.7px;
    margin-bottom: 5px;
  `,
  forSecurityText: css`
    margin-top: 1vh;
    display: block;
    color: #5b5b5b;
  `
};
