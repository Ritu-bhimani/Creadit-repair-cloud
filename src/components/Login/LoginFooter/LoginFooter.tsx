import { Container, Grid } from '@mui/material';
import { FC } from 'react';
import { useTranslation } from 'react-i18next';
import { LoginFooterStyles } from './styles';

type LoginFooterProps = {
  trainingLink: string;
  secureLoginImgSrc?: string;
};

export const LoginFooter: FC<LoginFooterProps> = (props: LoginFooterProps) => {
  const { trainingLink, secureLoginImgSrc } = props;

  // Translation hook to retrive the translations
  const { t } = useTranslation();

  return (
    <Container css={LoginFooterStyles.root}>
      <Grid container direction={'column'}>
        {/* Getting Started Section */}
        <div css={LoginFooterStyles.justGettingStartedText}>
          <span>
            {t('LoginText1')}
            <br />
            {t('LoginText2')}
          </span>
        </div>
        {/* Training Link */}
        <div>
          <a css={LoginFooterStyles.button} href={trainingLink}>
            {t('LoginText3')}
          </a>
        </div>
        {/* Security Information */}
        <div css={LoginFooterStyles.forSecurityText}>
          *For Security, User IDs and passwords cannot be shared.
          <br /> All users must have their own User ID, All visits are logged.
        </div>
        <div css={LoginFooterStyles.bottomText}>
          <img src={secureLoginImgSrc} />
          <span>Secure Area</span>
          This website is protected by 256-bit SSL security.
          <br />Â© 2012-2023 Credit Repair Cloud. All rights reserved. Patent
          Pending.
        </div>
      </Grid>
    </Container>
  );
};
