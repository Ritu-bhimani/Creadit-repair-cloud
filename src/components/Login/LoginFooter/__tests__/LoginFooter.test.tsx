import { render } from '@testing-library/react';
import { LoginFooter } from '../LoginFooter';

it('should render Login footer and match the snapshot', () => {
  const component = render(<LoginFooter trainingLink={''} />);
  expect(component).toMatchSnapshot();
});
