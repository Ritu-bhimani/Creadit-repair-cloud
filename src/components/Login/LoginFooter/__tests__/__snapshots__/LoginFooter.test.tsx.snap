// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Login footer and match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
        css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
        >
          <div
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          >
            <span>
              LoginText1
              <br />
              LoginText2
            </span>
          </div>
          <div>
            <a
              css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
              href=""
            >
              LoginText3
            </a>
          </div>
          <div
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          >
            *For Security, User IDs and passwords cannot be shared.
            <br />
             All users must have their own User ID, All visits are logged.
          </div>
          <div
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          >
            <img />
            <span>
              Secure Area
            </span>
            This website is protected by 256-bit SSL security.
            <br />
            © 2012-2023 Credit Repair Cloud. All rights reserved. Patent Pending.
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
      >
        <div
          css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
        >
          <span>
            LoginText1
            <br />
            LoginText2
          </span>
        </div>
        <div>
          <a
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
            href=""
          >
            LoginText3
          </a>
        </div>
        <div
          css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
        >
          *For Security, User IDs and passwords cannot be shared.
          <br />
           All users must have their own User ID, All visits are logged.
        </div>
        <div
          css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
        >
          <img />
          <span>
            Secure Area
          </span>
          This website is protected by 256-bit SSL security.
          <br />
          © 2012-2023 Credit Repair Cloud. All rights reserved. Patent Pending.
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
